client/state/sites/hooks/use-site-global-styles-status.ts

```diff
@@ -1,23 +1,58 @@
 import { useQuery } from '@tanstack/react-query';
 import wpcom from 'calypso/lib/wp';
+import { useSelector } from 'calypso/state';
+import { getSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
 export type GlobalStylesStatus = {
 	shouldLimitGlobalStyles: boolean;
 	globalStylesInUse: boolean;
+	globalStylesInPersonalPlan: boolean;
 };
 
 // While we are loading the Global Styles Info we can't assume that we should limit global styles, or we would be
 // showing notices for paid sites until we fetch the data from the server.
-export const DEFAULT_GLOBAL_STYLES_INFO: GlobalStylesStatus = {
+const DEFAULT_GLOBAL_STYLES_INFO: GlobalStylesStatus = {
 	shouldLimitGlobalStyles: false,
 	globalStylesInUse: false,
+	globalStylesInPersonalPlan: false,
 };
 
-export const getGlobalStylesInfoForSite = ( siteId: number | null ): GlobalStylesStatus => {
+function isObject( x: unknown ): x is Record< string, unknown > {
+	return typeof x === 'object' && x !== null;
+}
+
+const getExperimentAssignment = ( experimentName: string ): string | null => {
+	return wpcom.req
+		.get(
+			{
+				path: '/experiments/0.1.0/assignments/calypso',
+				apiNamespace: 'wpcom/v2/',
+			},
+			{
+				experiment_name: experimentName,
+			}
+		)
+		.then( ( response: unknown ) =>
+			isObject( response ) &&
+			isObject( response.variations ) &&
+			typeof response.ttl === 'number' &&
+			0 < response.ttl
+				? response.variations[ experimentName ]
+				: null
```

Comment:
This code has a multivalued return, it can either return `false`, `null`, or a `string`. Here is a proposal that only returns two types and is also a bit simpler:
```suggestion
			typeof response?.ttl === 'number' &&
			0 < response.ttl && response?.variations[ experimentName ] ?? false
```

Reply:
NVM you would need to define the response type, and since we'll be deleting this code eventually...